### PR TITLE
Update for new smartcard format

### PR DESCRIPTION
**Pull request by benadida**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -60,8 +67,14 @@ const App: React.FC = () => {
...
-            const cardData = JSON.parse(resultJSON.shortValue) as CardData
-            processCardData(cardData, resultJSON.longValueExists)
+            if (resultJSON.present) {
+              const cardData = JSON.parse(resultJSON.shortValue) as CardData
+              processCardData(cardData, resultJSON.longValueExists)
+            } else {
+              setIsWritableCard(false)
+            }
+          } else {
+            setIsWritableCard(true)
```
Comment (beausmith) [inline src/App.tsx]: I'm struggling to follow the logic in these nested if blocks. I think it's the `resultJSON.present`… does `present` mean "card is present in reader"? Renaming `resultJSON` to `card` will help.

Ah, the part that doesn't make sense: if `resultJSON.shortValue` exists, then won't the nested `resultJSON.present` always be true?

**Review by beausmith — CHANGES_REQUESTED** [review]
Most all this looks good… one part that doesn't make sense and for which I'm hoping can be made better.

File: src/App.tsx (hunk 1 above)
Comment (benadida) [reply src/App.tsx]: you're correct, the logic is slightly wrong here. I think the API still makes sense, you're reading the card, and it tells you whether it's present or not. But of course that inner `if` should be outside. I'll fix. Good catch

**Comment by benadida** [comment]
@beausmith that last commit is because something isn't working the way I expect with functional components and useState, and so the UI kept on polling the backend for the full election file. 

Tracking this here:
https://github.com/votingworks/bas/issues/15

File: src/App.tsx [hunk 2]
```diff
@@ -18,50 +18,71 @@ import './App.css'
 
 let checkCardInterval = 0
 
-// I want to use a hook for this, but it's not reacting quickly enough
+// I want to use a hook for these, but it's not reacting quickly enough
 // in the loop to prevent multiple loads of the long value.
 let loadingElection = false
+let electionLoaded = false
```
Comment (beausmith) [inline src/App.tsx:24]: Frustrated, as I think I know how to fix this, but I don't have a dev env which supports smart cards.

**Review by beausmith — APPROVED** [review]
LGTM… though acknowledging tech debt because I'm unable to use smartcards in my dev env.

**Pull request merged**